core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -85,6 +85,10 @@ private[spark] class Executor(
     env.blockManager.initialize(conf.getAppId)
   }
 
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
+  executorMetrics.setPort(env.blockTransferService.port)
```

Comment:
I don't think this is the "right" port -- eg., its not the port that users will see in the UI elsewhere for the executor.  I think that will be the port in `env.address.port` (that is what is used in the `onExecutorAdded` event).